filesystems/http/src/main/scala/cromwell/filesystems/http/HttpPathBuilder.scala

```diff
@@ -0,0 +1,46 @@
+package cromwell.filesystems.http
+import java.nio.file.Paths
+
+import akka.actor.{ActorContext, ActorSystem}
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model.HttpRequest
+import akka.stream.ActorMaterializer
+import akka.util.{ByteIterator, ByteString}
+import cromwell.core.path.{NioPath, Path, PathBuilder}
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Try}
+
+object HttpPathBuilder {
+  def accepts(url: String): Boolean = url.matches("^http[s]?://.*")
+}
+
+
+class HttpPathBuilder extends PathBuilder {
+  override def name: String = "HTTP"
+
+  override def build(pathAsString: String): Try[Path] = {
+    if (HttpPathBuilder.accepts(pathAsString)) Try {
+      HttpPath(Paths.get(pathAsString), this)
+    } else {
+      Failure(new IllegalArgumentException(s"$pathAsString does not have an http or https scheme"))
+    }
+  }
+
+  def contentAsByteIterator(input: String)(implicit actorContext: ActorContext, actorMaterializer: ActorMaterializer): Future[ByteIterator] = {
+    implicit val actorSystem: ActorSystem = actorContext.system
+    implicit val executionContext: ExecutionContext = actorContext.dispatcher
+    for {
+      response <- Http().singleRequest(HttpRequest(uri = input))
+      byteString <- response.entity.dataBytes.runFold(ByteString(""))(_ ++ _)
+    } yield byteString.iterator
```

Comment:
Just to double check, I may be wrong but this looks like it downloads the entire file before creating an iterator out of it?

ie if someone does `read_string(oops.bam)` will we download the whole thing to memory, produce an iterator for it, and then hit the "file too big" only while iterating through the in-memory iterator?